### PR TITLE
Use auto table-layout on desktop

### PR DESCRIPTION
**Pull request by not-my-profile**
On desktop the tables in the specification currently have quite awkward column widths, making them harder to read.

| Before | After |
|--|--|
|image | image

While the fixed table-layout does have an advantage on mobile since it avoids unnecessary horizontal scrolling in the tables, we can just use it on narrow browser widths via a media query, while still using the more readable auto layout on desktop.

Preview: https://pr1601--matrix-spec-previews.netlify.app

### CODE REVIEW

File: assets/scss/custom.scss [hunk 1]
```diff
@@ -350,19 +350,37 @@ footer {
...
+
+       * Docsy makes all tables "responsive tables", which causes Bootstrap 4 to create
+       * tables with a "display" property of "block".
+       * However, for "table-layout: fixed" to be effective, an element must have a
+       * "display" property of "table".
+       *
+       * Thus, we override the "display" property here. This may no longer be necessary once
+       * Docsy updates to Bootstrap v5+: https://github.com/google/docsy/issues/470.
+       * For more details, see
+       * https://github.com/matrix-org/matrix-spec/pull/1295/files#r1010759688 */
```
Comment (richvdh) [inline assets/scss/custom.scss:367]: I wonder if we should update our fork of docsy now that https://github.com/google/docsy/issues/470 is closed...

Anyway, I guess that is a separate problem.

**Review by richvdh — CHANGES_REQUESTED** [review]
Thanks! Please add a changelog and sign-off.

**Comment by not-my-profile** [comment]
Just signed-off the commit. I'd be happy to add a changelog entry but none of the allowed changelog entry types fits this PR (new, feature, clarification, breaking, deprecation).  I'd classify this as "style" but I'm assuming `scripts/check-newsfragments` won't let that pass.

**Comment by richvdh** [comment]
Thanks! You could make it a `clarification` in the [internal](https://github.com/matrix-org/matrix-spec/tree/main/changelogs/internal/newsfragments) section.

**Comment by not-my-profile** [comment]
CONTRIBUTING.rst defines `clarification` as:

> clarification - Used when an area of the spec is being improved upon and does not change or introduce any functionality.

That really sounds like a change to the text of the spec ... so I don't really want to classify it as "clarification" since I'd consider that as misleading.

**Comment by richvdh** [comment]
I mean you're right, and https://github.com/matrix-org/matrix-spec/issues/1369 mentions the fact that it's not a great match. But it's what we use for now.

Compare https://spec.matrix.org/v1.7/changelog/#internal-changestooling for where we've done this in the past. None of those are really changes to the spec.

**Comment by not-my-profile** [comment]
Fair enough ... added the newsfragment.

File: assets/scss/custom.scss [hunk 2]
```diff
@@ -350,19 +350,37 @@ footer {
...
-     * https://github.com/matrix-org/matrix-spec/pull/1295/files#r1010759688 */
-    display: table;
-    table-layout: fixed;
-    width: 100%;
+    @media (max-width: 800px) {
+      /* Docsy by default applies `overflow-x: auto;` to tables, which
+       * results in annoying horizontal scrolling on mobile, so we instead
+       * switch to a fixed table layout on a narrow browser width.
+       * (On a wider width the default auto table-layout provides better readability.)
+
```
Comment (richvdh) [inline assets/scss/custom.scss:358]:
```suggestion
      *
```

File: changelogs/internal/newsfragments/1601.clarification [hunk 3]
```diff
@@ -0,0 +1 @@
+Improve the layout of tables on desktop.
```
Comment (richvdh) [inline changelogs/internal/newsfragments/1601.clarification:1]:
```suggestion
Improve the layout of tables on desktop displays. Contributed by @not-my-profile.
```

**Review by richvdh — APPROVED** [review]
Thank you!

(Aside: it's fine in this case, because the changes are relatively small, but please don't force-push after a review: it makes it hard to see what has changed since the previous review. We can clean up the commit history when we merge the PR.)

File: assets/scss/custom.scss [hunk 4]
```diff
@@ -350,19 +350,37 @@ footer {
...
-     * https://github.com/matrix-org/matrix-spec/pull/1295/files#r1010759688 */
-    display: table;
-    table-layout: fixed;
-    width: 100%;
+    @media (max-width: 800px) {
+      /* Docsy by default applies `overflow-x: auto;` to tables, which
+       * results in annoying horizontal scrolling on mobile, so we instead
+       * switch to a fixed table layout on a narrow browser width.
+       * (On a wider width the default auto table-layout provides better readability.)
+      *
```
Comment (richvdh) [inline assets/scss/custom.scss:358]: damnit

```suggestion
       *
```

File: changelogs/internal/newsfragments/1601.clarification (hunk 3 above)
Comment (not-my-profile) [reply changelogs/internal/newsfragments/1601.clarification]: I don't really like my work being attributed to my GitHub username :/

Comment (richvdh) [reply changelogs/internal/newsfragments/1601.clarification]: Oh, sorry. I can remove this.

Comment (richvdh) [reply changelogs/internal/newsfragments/1601.clarification]: would you prefer an alternative attribution?

Comment (not-my-profile) [reply changelogs/internal/newsfragments/1601.clarification]: Yeah: Martin Fischer.

Comment (richvdh) [reply changelogs/internal/newsfragments/1601.clarification]: Done: https://github.com/matrix-org/matrix-spec/commit/f5035b8e0271649816ab35dc12e9a5cc5e7783c3

Comment (not-my-profile) [reply changelogs/internal/newsfragments/1601.clarification]: Thanks <3

**Pull request merged**